wordpress-mobile/WordPress-Android
# Fix/login ui tests

Part of #9296 

This PR aims to get the tests originally written in https://github.com/wordpress-mobile/WordPress-Android/compare/add/login-ui-tests and up and running again. 

What changed:
* Stored tests in an e2e folder
* The Page object structure a little
* Moved support classes one level up

To test:

Run tests in e2e folder on a phone emulator, all tests should pass. 

**NB**

The screenshot tests fail for me with 

`
java.lang.RuntimeException: Unable to capture screenshot.
at tools.fastlane.screengrab.FileWritingScreenshotCallback.screenshotCaptured(FileWritingScreenshotCallback.java:56)
at tools.fastlane.screengrab.UiAutomatorScreenshotStrategy.takeScreenshot(UiAutomatorScreenshotStrategy.java:39)
at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:94)
at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:76)
at tools.fastlane.screengrab.Screengrab.screenshot(Screengrab.java:60)
at org.wordpress.android.ui.screenshots.WPScreenshotTest.editBlogPost(WPScreenshotTest.java:96)
at org.wordpress.android.ui.screenshots.WPScreenshotTest.wPScreenshotTest(WPScreenshotTest.java:63)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at android.support.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:527)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.wordpress.android.ui.screenshots.WPLocaleTestRule$1.evaluate(WPLocaleTestRule.java:37)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at android.support.test.runner.AndroidJUnit4.run(AndroidJUnit4.java:101)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at android.support.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at android.support.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:384)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
Caused by: java.io.IOException: Unable to get a screenshot storage directory
at tools.fastlane.screengrab.FileWritingScreenshotCallback.getFilesDirectory(FileWritingScreenshotCallback.java:81)
at tools.fastlane.screengrab.FileWritingScreenshotCallback.screenshotCaptured(FileWritingScreenshotCallback.java:39)
... 38 more
`

but I think that's an unrelated error, was there something else I needed to setup for that to get those going?
